### PR TITLE
Feature/ingame dialogue

### PR DESCRIPTION
**Pull request by baconeta**
This adds the in game dialogue system to the game - everything except the actual conversations data which will come in a future PR

### CODE REVIEW

File: Assets/Scripts/UI/BarksController.cs [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        GameStateManager.OnPauseEnter += StopBarkTimers;
+        GameStateManager.OnPauseExit += StartBarkTimers;
+        DialogueManager.OnDialogueStart += CloseAllPopups;
+    }
+
+    private void OnDisable()
+    {
+        BoatController.OnDamageTaken -= HitSomething;
+        BoatController.OnVoyageStart -= StartBarkTimers;
+        BoatController.OnVoyageComplete -= StartBarkTimers;
```
Comment (TomasMorton) [inline Assets/Scripts/UI/BarksController.cs]: This one should be StopBarkTimers

Comment (baconeta) [reply Assets/Scripts/UI/BarksController.cs]: Thanks, good catch

File: Assets/Scripts/UI/BarksController.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    private void SetRightSpeaker(string speaker, string bark)
+    {
+        barkPopupRight.SetSpeaker(speaker);
+        barkPopupRight.SetText(bark);
+        barkPopupRight.SetSpeakerPic(GetSpeakerSprite(speaker));
+    }
+
+    private float CalculateTimeOnScreen(string text)
+    {
+        return 3f;
```
Comment (TomasMorton) [inline Assets/Scripts/UI/BarksController.cs:101]: Superb calculation!

**Comment by TomasMorton** [comment]
Probably needs some tuning, it's very hard (impossible?) to get a hit bark because of the cooldown + how quickly a new bark appears

**Review by TomasMorton — APPROVED** [review]

**Pull request merged**